sdk/src/invariant.ts

```diff
@@ -0,0 +1,84 @@
+import { ApiPromise } from "@polkadot/api";
+import { ContractPromise } from "@polkadot/api-contract";
+import { WeightV2 } from "@polkadot/types/interfaces";
+import { IKeyringPair } from "@polkadot/types/types/interfaces";
+import { deployContract } from "@scio-labs/use-inkathon/helpers";
+
+export class Invariant {
+  contract: ContractPromise | null = null;
+  api: ApiPromise;
+  account: IKeyringPair;
+  weight: WeightV2;
+
+  constructor(
+    api: ApiPromise,
+    account: IKeyringPair,
+    refTime: number,
+    proofSize: number
+  ) {
+    this.api = api;
+    this.account = account;
+    this.weight = api.registry.createType("WeightV2", {
+      refTime,
+      proofSize,
+    }) as WeightV2;
+  }
+
+  async new(abi: any, wasm: Buffer, fee: { v: number }): Promise<void> {
+    const invariant_deployment = await deployContract(
+      this.api,
+      this.account,
+      abi,
+      wasm,
+      "new",
+      [fee]
+    );
+    this.contract = new ContractPromise(
+      this.api,
+      abi,
+      invariant_deployment.address
+    );
+  }
+
+  async getProtocolFee(): Promise<void> {
+    if (!this.contract) {
```

Comment:
Is there possibility to abstract query similarly to this: https://github.com/invariant-labs/psp22-ts/blob/add-transfer-and-metadata-messages/src/token.ts#L58